### DISCUSSION TITLE
Configure cgroups mounts as shared and mount bpffs by default

### DISCUSSION THREAD

**Discussion by chancez** [rancher-sandbox/rancher-desktop#1977] [Ideas]
I work for @isovalent and so I'm always looking for ways to make installing cilium easier. I've been making some progress on that in #1976 for example. 

One thing that would make it even easier to install cilium is making `/sys/fs/cgroup` a shared mount, and mounting the BPF at `/sys/fs/bpf` and also making it a shared mount. This is necessary because the cilium-agent container mounts these using host mounts, and it requires them to be shared mounts for this work.

Today I've accomplished this with a `override.yaml` for Lima:

```
provision:
  - mode: system
    script: |
      #!/bin/sh
      set -e

      # needed for cilium
      mount bpffs -t bpf /sys/fs/bpf
      mount --make-shared /sys/fs/bpf
      mount --make-shared /sys/fs/cgroup
```

This works well, but requires providing additional instructions for users, and it doesn't work for WSL.

The other benefit of this is that it would enable using BPF on rancher-desktop with other BPF tools besides cilium, that run in containers, so it's not entirely cilium specific.

**Comment by jandubois** [comment]
> One thing that would make it even easier to install cilium is making `/sys/fs/cgroup` a shared mount, and mounting the BPF at `/sys/fs/bpf` and also making it a shared mount.

This sounds fine to me. It could just be added to the RD provisioning scripts (and `wsl-init` for Windows). Just file a PR for it!

**Reply by chancez** [reply]
Sounds good to me!
